ONSdigital/rm-action-service
# Sts008 receive notification letter

The caseRef was already passed through to the action service this change is setting the caseRef so that it can be used in the pre notification file. 
This change also adds new action types into the action table required for social surveys

**How to test:**

1. Run this along with actionexporter in docker
2. Run `make setup` in the acceptance tests
3. Go into the `action.actionrule` table and you'll need to create a new rule
4. Create a 5th column and add this data to the column - 2, 3, 4, BRICKSOCNOT, Bricks social notification, 10000, 3, b1232e28-7268-481b-b0d0-b4cd8c49e5b1

Like this below:

![screen shot 2018-06-07 at 15 31 19](https://user-images.githubusercontent.com/16239750/41106415-e3ba332c-6a67-11e8-8ee9-9e6a4eb2d87c.png)

5. Load a sample and CI manually for Bricks 201801 while tailing the logs of the action, actionexporter and collection exercise services. You should ultimately see the CSV being created in the action exporter.

6. Set the collection exercise ready for live

7. Go back to the database in the `action.actionrule` table and set the `daysoffset` back to 0 and save

8. Check the contents of the CSV and see the `caseRef` at the end of the CSV file:

Get the social file name and check it exists with:

`Docker exec sftp ls -lrt /home/centos/Documents/sftp` 

Then plug that social file csv name into this command to check the caseRef is added into the csv:

`Docker exec sftp cat /home/centos/Documents/sftp/SOCIALFILENAMEHERE.csv`

The address columns will be blank since the loaded sample will be a business sample that does not contain address information.

# Links
https://github.com/ONSdigital/rm-actionsvc-api/pull/14
https://github.com/ONSdigital/rm-actionexporter-service/pull/26
https://trello.com/c/xbFrDWWf/100-dev-task-5-int-notify-respondent-of-invitation-to-a-survey-5
